File: 2026/06-June/14-Sunday/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

int pairSum(struct ListNode* head) {
    struct ListNode *slow = head, *fast = head;

    while (fast && fast->next) {
        slow = slow->next;
        fast = fast->next->next;
    }

    struct ListNode *prev = NULL, *curr = slow;
    while (curr) {
        struct ListNode *nextNode = curr->next;
        curr->next = prev;
        prev = curr;
        curr = nextNode;
    }

    int ans = 0;
    struct ListNode *first = head;
    struct ListNode *second = prev;

    while (second) {
        int sum = first->val + second->val;
        if (sum > ans) ans = sum;
        first = first->next;
        second = second->next;
    }

    return ans;
}

File: 2026/06-June/14-Sunday/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func pairSum(head *ListNode) int {
    slow, fast := head, head

    for fast != nil && fast.Next != nil {
        slow = slow.Next
        fast = fast.Next.Next
    }

    var prev *ListNode
    curr := slow

    for curr != nil {
        nextNode := curr.Next
        curr.Next = prev
        prev = curr
        curr = nextNode
    }

    ans := 0
    first := head
    second := prev

    for second != nil {
        if first.Val+second.Val > ans {
            ans = first.Val + second.Val
        }
        first = first.Next
        second = second.Next
    }

    return ans
}
